Fix failing 'Swiftlint' and 'Run Tests' Github Actions workflows

martinrist opened the pull request:
The 'SwiftLint' workflow is currently failing in GitHub Actions, with the the following error:

```
Error: The `brew link` step did not complete successfully
The formula built, but is not symlinked into /usr/local
Could not symlink bin/swiftlint
Target /usr/local/bin/swiftlint
already exists. You may want to remove it:
  rm '/usr/local/bin/swiftlint'

To force the link and overwrite all conflicting files:
  brew link --overwrite swiftlint

To list all files that would be deleted:
  brew link --overwrite --dry-run swiftlint

Possible conflicting files are:
/usr/local/bin/swiftlint
==> Summary
🍺  /usr/local/Cellar/swiftlint/0.43.0: 6 files, 13.4MB
Error: Process completed with exit code 1.
```

The exit code of '1' then causes the workflow to fail.

The root cause of this failure is that, from [`macos-10.14`](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.14-Readme.md) onwards, SwiftLint is installed in the base image used by 
the workflow.  Attempting to install SwiftLint using `brew` then causes the errors seen above.

This PR simply removes the 'Install SwiftLint' step, so that we just rely on the underlying version of SwiftLint installed on the base image.

[Edit]: In addition, this PR adjusts some of the expectation wait timeouts, to correct additional failing tests in the 'Run tests' workflow.

martinrist committed:
Remove 'Install SwiftLint' step

SwiftLint is installed in GitHub's base virtual environment image since
[`macos-10.14`](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.14-Readme.md).

Attempting to install SwiftLint using `brew` was causing this workflow
to fail with the following errors, since `brew` was unable to overwrite
the `swiftlint` symlink already installed in the image:

```
Error: The `brew link` step did not complete successfully
...
Error: Process completed with exit code 1.
```

Note: this used to work because the workflow is using `macos-latest`.
This used to resolve to
[`macos-10.13`](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-10.13-Readme.md)
which didn't include SwiftLint in the base image.

martinrist commented:
I've realised that this PR won't be able to be merged until the 'Run tests' workflow also passes.

Although the tests are all passing locally, some of them are failing because some of the tests in `PersistenceStore_DownloadsTest` and `DownloadServiceTest` are exceeding the 5-second timeouts specified in various places (e.g. [here](https://github.com/razeware/emitron-iOS/blob/8225da4142db91b9a5606a422d824fc711bc9f4f/Emitron/emitronTests/Persistence/PersistenceStore%2BDownloadsTest.swift#L70) and [here](https://github.com/razeware/emitron-iOS/blob/8225da4142db91b9a5606a422d824fc711bc9f4f/Emitron/emitronTests/Downloads/DownloadServiceTest.swift#L333).

Is it acceptable to bump some of these timeouts within the scope of this PR?

0xTim commented:
@martinrist Yeah since this PR is about fixing CI I'm happy to bundle increasing the timeouts onto it

martinrist committed:
Increase wait timeout to fix CI

A number of tests have been failing on CI for recent PRs.  This commit
increases the timeout for tests that have been affected on the most
recent PRs.

martinrist commented:
I've reviewed the recently-failing workflow runs, and have adjusted the timeouts on the affected tests, so I now have a passing build.  I'm happy to adjust all the other timeouts as well if it's felt that would be appropriate at this stage.

I've adjusted this PR's title and description to include the scope change.

sammyd reviewed (approved):
:+1: Thanks for this.

I think this is good to merge—there's nothing controversial here, but maybe @0xTim will correct me after the fact...

0xTim commented:
Nope this was all good on my end
